libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/voice/AndroidSpeechPlayer.java

```diff
@@ -33,26 +36,39 @@
     textToSpeech = new TextToSpeech(context, new TextToSpeech.OnInitListener() {
       @Override
       public void onInit(int status) {
-        if (status != TextToSpeech.ERROR) {
-          textToSpeech.setLanguage(new Locale(language));
+        boolean ableToInitialize = status != TextToSpeech.ERROR && language != null;
+
+        if (!ableToInitialize) {
+          Timber.e("There was an error initializing native TTS");
         }
+
+        initializeWithLanguage(new Locale(language));
+
       }
     });
   }
 
   /**
+   * Plays the given voice instruction using TTS
+   *
    * @param instruction voice instruction to be synthesized and played
    */
   @Override
   public void play(String instruction) {
-    if (!isMuted && !TextUtils.isEmpty(instruction)) {
-      HashMap<String, String> params = new HashMap<>(1);
-      params.put(TextToSpeech.Engine.KEY_PARAM_UTTERANCE_ID, DEFAULT_UTTERANCE_ID);
-      textToSpeech.speak(instruction, TextToSpeech.QUEUE_ADD, params);
+    boolean canPlay = languageSupported && !isMuted && !TextUtils.isEmpty(instruction);
```

Comment:
For readability, what about negating the condition and calling the boolean `cannotPlay`?